### PR TITLE
backports from master (part 1)

### PR DESCRIPTION
**Pull request by Dikay900**
sample references( #81 #3229 #823 #3217 #785 #841 #3867 #3615 #2850 #3925)
referencing needed backports #3906 #3744 #3729

edit: reordered PR list to commits
edit2: if needed i can also split this PR in more PRs please give then a max amount of commits to be used at once for one PR

### CODE REVIEW

**Comment by Dikay900** [comment]
added backports for #3906 #3744 #3729

**Comment by Dikay900** [comment]
if referenced people are active please review your own commits, if someone find a bug please give me a feedback.

**Comment by StevenPuttemans** [comment]
:+1:

File: samples/cpp/openni_capture.cpp [hunk 1]
```diff
@@ -12,14 +12,14 @@ static void help()
                         "The user gets some of the supported output images.\n"
             "\nAll supported output map types:\n"
             "1.) Data given from depth generator\n"
-            "   OPENNI_DEPTH_MAP            - depth values in mm (CV_16UC1)\n"
-            "   OPENNI_POINT_CLOUD_MAP      - XYZ in meters (CV_32FC3)\n"
-            "   OPENNI_DISPARITY_MAP        - disparity in pixels (CV_8UC1)\n"
-            "   OPENNI_DISPARITY_MAP_32F    - disparity in pixels (CV_32FC1)\n"
-            "   OPENNI_VALID_DEPTH_MASK     - mask of valid pixels (not ocluded, not shaded etc.) (CV_8UC1)\n"
+            "   CV_CAP_OPENNI_DEPTH_MAP            - depth values in mm (CV_16UC1)\n"
```
Comment (StevenPuttemans) [inline samples/cpp/openni_capture.cpp]: Why would we reintroduce the CV_ prefixes in master? Seems invalid ...

Comment (Dikay900) [inline samples/cpp/openni_capture.cpp]: this is the backport pr to 2.4 where these constants still exists in this format

Comment (StevenPuttemans) [inline samples/cpp/openni_capture.cpp]: o wow :D confused here :D

Comment (Dikay900) [inline samples/cpp/openni_capture.cpp]: sry for that :blush: shame on me changed that for future prs

**Comment by vpisarev** [comment]
:+1:

**Pull request merged**